src/MeshKernel/CurvilinearGrid/CurvilinearGridSmoothing.cpp

```diff
@@ -63,16 +63,16 @@ CurvilinearGrid CurvilinearGridSmoothing::ComputeDirectional()
     }
 
     // Points are coinciding, this no smoothing zone
-    if (m_lines[0].m_gridLineType == CurvilinearGridLine::GridLineType::MGridLine && m_lowerLeft.m_n == m_upperRight.m_n ||
-        m_lines[0].m_gridLineType == CurvilinearGridLine::GridLineType::NGridLine && m_lowerLeft.m_m == m_upperRight.m_m)
+    if (m_lines[0].m_gridLineType == CurvilinearGrid::GridLineDirection::MDirection && m_lowerLeft.m_n == m_upperRight.m_n ||
+        m_lines[0].m_gridLineType == CurvilinearGrid::GridLineDirection::NDirection && m_lowerLeft.m_m == m_upperRight.m_m)
     {
         throw std::invalid_argument("CurvilinearGridSmoothing::Compute The points defining the smoothing area have the same direction of the smoothing line.");
     }
 
     // Re-compute the smoothing block: the block must coincide with the line end and start nodes
     CurvilinearGrid::NodeIndices lowerLeftBlock;
     CurvilinearGrid::NodeIndices upperRightBlock;
-    if (m_lines[0].m_gridLineType == CurvilinearGridLine::GridLineType::MGridLine)
+    if (m_lines[0].m_gridLineType == CurvilinearGrid::GridLineDirection::MDirection)
```

Comment:
since you do this check a lot, you could add a member function to CurvilinearGridLine:
bool CurvilinearGridLine::isMDirectionLine(){return m_gridLineType==GridLineDirection::MDirection;)